src/interfaces/background_field_interface.jl

```diff
@@ -0,0 +1,198 @@
+
+The only supported class of background fields yet, are the pulsed plane-wave field (see [`AbstractPulsedPlaneWaveField`](@ref) for details).
+"""
+abstract type AbstractBackgroundField end
+
+#####
+# pulsed plane-wave fields
+#####
+
+"""
+Abstract base type for pulsed plane-wave fields.
+
+!!! note "Pulsed field interface"
+
+    For every subtype, the following functions need to be implemented.
+
+    ```Julia
+
+    reference_momentum(::AbstractPulsedPlaneWaveField)
+    domain(::AbstractPulsedPlaneWaveField)
+    phase_duration(::AbstractPulsedPlaneWaveField)
+    envelope(::AbstractPulsedPlaneWaveField,x::Real)
+    ```
+
+"""
+abstract type AbstractPulsedPlaneWaveField <: AbstractBackgroundField end
+
+"""
+
+    reference_momentum(::AbstractPulsedPlaneWaveField)
+
+Interface function for [`AbstractPulsedPlaneWaveField`](@ref), which returns the reference momentum (as a subtype of `QEDbase.AbstractFourMomentum`) associated with the passed background field.
+"""
+function reference_momentum end
+
+"""
+    domain(::AbstractPulsedPlaneWaveField)
+
+Interface function for [`AbstractPulsedPlaneWaveField`](@ref), which returns interval (as a `IntervalSets.Interval`) for the given background field. 
+
+"""
+function domain end
+
+"""
+    
+    phase_duration(::AbstractPulsedPlaneWaveField)
+
+Interface function for [`AbstractPulsedPlaneWaveField`](@ref), which returns the phase extent of the background field.
+"""
+function phase_duration end
+
+"""
+
+    _envelope(::AbstractPulsedPlaneWaveField, phi::Real)
+
+Interface function for [`AbstractPulsedPlaneWaveField`](@ref), which returns the value of the phase envelope function (also referred to as pulse envelope) for a given phase variable.
+
+!!! note "Single point implementation"
+
+    The interface function can be implemented for just one phase point as input. With that, evaluation on a vector of inputs is generically implemented by broadcasting.
+    However, if there is a better custom implementation for vectors in input values, consider implementing 
+    ```Julia
+    
+        _envelope(::AbstractPulsedPlaneWaveField, phi::AbstractVector{T<:Real})
+
+    ```
+
+!!! note "unsafe implementation"
+    
+    This is the unsafe version of the phase envelope function, i.e. this should be implement without input checks like the domain check. 
+    In the safe version [`envelope`](@ref), a domain check is performed, i.e. it returns the value of `_envelope` if the passed in `phi` 
+    is in the `domain` of the field, and zero otherwise. 
+
+"""
+function _envelope end
+
+function _envelope(
+    field::AbstractPulsedPlaneWaveField, phi::AbstractVector{T}
+) where {T<:Real}
+    # TODO: maybe use broadcasting here 
+    return map(x -> _envelope(field, x), phi)
+end
+
+"""
+    
+    envelope(pulsed_field::AbstractPulsedPlaneWaveField, phi::Real)
+    
+Return the value of the phase envelope funtion (also referred to as pulse envelope or pulse shape) 
```

Comment:
typo function

Reply:
Seems like it is outdated?

Reply:
I still see it in the code (https://github.com/szabo137/QEDfields.jl/blob/224e6d8fe2ec7e5781de7e8d2cf672cbd4ebbafa/src/interfaces/background_field_interface.jl#L97)

Reply:
Fixed in 73f7bfc